utils/src/main/scala/za/co/absa/enceladus/utils/types/TypedStructField.scala

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.utils.types
+
+import org.apache.spark.sql.types._
+import za.co.absa.enceladus.utils.implicits.StructFieldImplicits.StructFieldEnhancements
+import za.co.absa.enceladus.utils.schema.MetadataKeys
+import za.co.absa.enceladus.utils.time.DateTimePattern
+import za.co.absa.enceladus.utils.types.parsers._
+import za.co.absa.enceladus.utils.validation.ValidationIssue
+import za.co.absa.enceladus.utils.validation.field.{DateFieldValidator, DateTimeFieldValidator, FieldValidator, ScalarFieldValidator, TimestampFieldValidator}
+
+import scala.util.{Failure, Success, Try}
+
+sealed abstract class TypedStructField(structField: StructField) extends StructFieldEnhancements(structField) {
+  protected def convertString(string: String): Try[Any]
+
+  def validate():Seq[ValidationIssue]
+
+  def stringToTyped(string: String): Try[Option[Any]] = {
+    if (string == null) {
+      if (structField.nullable) {
+        Success(None)
+      } else {
+        Failure(new IllegalArgumentException(s"null is not a valid value for field '${structField.name}'"))
+      }
+    } else {
+      convertString(string).map(Some(_))
+    }
+  }
+
+  def ownDefaultValue: Try[Option[Option[Any]]] = {
```

Comment:
Is there maybe a way to make the return value less complicated?

Reply:
Very interesting

Reply:
I am marking this as resolved. I presume no one has any more comments here.